src/renderer/store/modules/profiles.js

```diff
@@ -89,6 +89,33 @@ const actions = {
     commit('setProfileList', profiles)
   },
 
+  async updateChannelThumbnail({ getters, dispatch }, { channelThumbnailUrl, channelName, channelId }) {
+    const profileList = getters.getProfileList
+    for (let i = 0; i < profileList.length; i++) {
+      const currentProfile = JSON.parse(JSON.stringify(profileList[i]))
+      const channelIndex = currentProfile.subscriptions.findIndex((channel) => {
+        return channel.id === channelId
+      })
+      if (channelIndex !== -1) {
+        console.log('channel found')
+        let updated = false
+        if (profileList[i].subscriptions[channelIndex].name !== channelName) {
```

Comment:
Why do you need to read from `profileList[i].subscriptions[channelIndex]` when you read from `currentProfile.subscriptions[channelIndex]`?